### PR TITLE
Adjusts minesweeper rewards & syndicate mines

### PR DESCRIPTION
**Pull request by cowbot92**
# Document the changes in your pull request

Adjusts minesweeper rewards

Removed rocket launcher & rockets from reward pool, replaced with syndicate mines (they looks like ducks)
Adjusted the sound of the syndicate mine exploding to be a quack
Adjusted it so you can only win one emagged minesweeper reward per machine
Adjusts it so when emagging the minesweeper machine, the difficulty will be force set to expert

Adjusts it so if you attempt to take apart an emagged machine/destroy it, it will explode to deter you from doing so.

# Why is this good for the game?

you know what you did

# Testing

it works @Chubbygummibear assisted with testing

# Changelog

:cl:  Cowbot92 & Chubbygummibear & Mqiib

tweak: adjusts how emagged minesweeper rewards/works
tweak: adjusts the sound of syndicate mines

/:cl:

### CODE REVIEW

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -243,12 +243,12 @@
 
 /obj/effect/mine/sound
 	name = "honkblaster 1000"
-	var/sound = 'sound/items/bikehorn.ogg'
+	var/sound = 'yogstation/sound/misc/quack.ogg'
 	disarm_time = 60 SECONDS // very long disarm time to expand the annoying factor
 	disarm_product = /obj/item/deployablemine/honk
```
Comment (Moltijoe) [inline code/game/objects/effects/mines.dm:248]: Honk mine from the quack sound effect?

Comment (Mqiib) [reply code/game/objects/effects/mines.dm:248]: it's a duck

Comment (cowbot92) [reply code/game/objects/effects/mines.dm:248]: its a duck

**Comment by ynot01** [comment]
What prevents someone from disassembling and reassembling a minesweeper game to get more items, circumventing the "one reward per machine" logic?

**Comment by PibeCantina** [comment]
I dont like this because this is much of a neft, imo, if you wanna set it to expert, at least keep the juiciest rewards in, minesweeper on expert is hard even if you know what you are doing, you are at the mercy of rng.

**Comment by cowbot92** [comment]
> What prevents someone from disassembling and reassembling a minesweeper game to get more items, circumventing the "one reward per machine" logic?

im cooking

**Comment by cowbot92** [comment]
> I dont like this because this is much of a neft, imo, if you wanna set it to expert, at least keep the juiciest rewards in, minesweeper on expert is hard even if you know what you are doing, you are at the mercy of rng.

sorry you have to work for your free traitor items now

**Comment by ShoziOffical** [comment]
My gimmicks gon

**Comment by MajManatee** [comment]
now they might have to make multiple machines to get free TC items

**Comment by cowbot92** [comment]
> What prevents someone from disassembling and reassembling a minesweeper game to get more items, circumventing the "one reward per machine" logic?

with my latest push I have updated it so that if you attempt to take apart a machine with a screwdriver while it is emagged, or destroy it, it will explode to deter people from doing this,

File: code/modules/modular_computers/file_system/programs/minesweeper.dm [hunk 2]
```diff
@@ -335,7 +347,7 @@
 		if(MINESWEEPER_EXPERT) // 30x16, 99 mines
 			width = 30
 			height = 16
-			mines = 99
+			mines = 0
```
Comment (Runian) [inline code/modules/modular_computers/file_system/programs/minesweeper.dm]: 99 -> 0? seems accidental

File: code/modules/modular_computers/file_system/programs/minesweeper.dm [hunk 3]
```diff
@@ -286,38 +291,45 @@
...
+	
+	if(result)
+		time_frozen = REALTIMEOFDAY - starting_time
+
+	return TRUE
+
+/datum/minesweeper/proc/winner_chicken_dinner(mob/user)
+	play_snd('yogstation/sound/arcade/minesweeper_win.ogg')
+	host.say("You cleared the board of all mines! Congratulations!")
+	if(emaggable && host.obj_flags & EMAGGED && value >= 1 && !emagwin)
```
Comment (Runian) [inline code/modules/modular_computers/file_system/programs/minesweeper.dm:310]: non-emag winners do not appear to get any ticket_count anymore. accidental?

File: code/modules/modular_computers/file_system/programs/minesweeper.dm (hunk 2 above)
Comment (cowbot92) [reply code/modules/modular_computers/file_system/programs/minesweeper.dm]: LOL forgot my fucking testing unit

Comment (cowbot92) [reply code/modules/modular_computers/file_system/programs/minesweeper.dm]: Fixed w/ latest push

File: code/modules/modular_computers/file_system/programs/minesweeper.dm (hunk 3 above)
Comment (cowbot92) [reply code/modules/modular_computers/file_system/programs/minesweeper.dm:310]: Fixed with latest push

**Comment by 00ze-cyclone** [comment]
This still fix nothing from it being infinite uplink item dispenser (albeit with an actual risk now), arcade machines are dirt cheap to make, making them explode just let you now create explosive walls which tbh is interesting.
We should just make emag not provide items with an uplink entry instead of those roundabout bandaids. Why pay TCs, your most precious ressource when you can get it with an emag.

**Comment by cowbot92** [comment]
> This still fix nothing from it being infinite uplink item dispenser (albeit with an actual risk now), arcade machines are dirt cheap to make, making them explode just let you now create explosive walls which tbh is interesting. We should just make emag not provide items with an uplink entry instead of those roundabout bandaids. Why pay TCs, your most precious ressource when you can get it with an emag.

It will make it so it is more of a challenge to get these items. They are still intended to get for the risk/reward, if I wanted to remove them I easily could have.

File: code/game/objects/effects/mines.dm (hunk 1 above)
Comment (Moltijoe) [reply code/game/objects/effects/mines.dm:248]: then why isn't it giving the duck mine back?

Comment (MajManatee) [reply code/game/objects/effects/mines.dm:248]: hes true you know

Comment (cowbot92) [reply code/game/objects/effects/mines.dm:248]: oh wait this one wasnt a duck, fixed it

**Comment by neeshacark** [comment]
i think this would probably be fine with just the 'one reward per machine' change and expert mode force, the issue has always been people sitting at a machine and getting insane amounts of stuff.  rocket launcher also isnt that good from my memory so removing it just seems kinda odd especially with the other changes

**Comment by cowbot92** [comment]
> i think this would probably be fine with just the 'one reward per machine' change and expert mode force, the issue has always been people sitting at a machine and getting insane amounts of stuff. rocket launcher also isnt that good from my memory so removing it just seems kinda odd especially with the other changes

the rocket launcher is in fact that good.

**Review by Chubbygummibear — APPROVED** [review]
lgtm

**Comment by MajManatee** [comment]
the balance review in question

**Pull request merged**